src/types.ts

```diff
@@ -1,10 +1,73 @@
-import { MouseEvent } from 'react'
-import { ItemProps } from './item'
+import { MouseEvent, MutableRefObject, ReactNode } from 'react'
+import PhotoSwipe, { PhotoSwipeOptions, UIElementData } from 'photoswipe'
+import PhotoSwipeLightboxStub from './lightbox-stub'
 
-export type ItemRef = React.MutableRefObject<HTMLElement>
+/**
+ * At Gallery level we can freely assume that ref is HTMLElement since we don't use any of html attributes.
+ * At Item level we can either set ref type like <Item<HTMLImageElement>> to ensure typing or simply omit it since
+ * elements like HTMLImageElement extends HTMLElement.
+ */
+export type ItemRef<NodeType extends HTMLElement = HTMLElement> =
+  MutableRefObject<NodeType | null>
 
-export type InternalItem = Omit<ItemProps, 'children'>
+export interface InternalItem {
+  /**
+   * Url of original image
+   */
+  original?: string
 
+  /**
+   * Srcset of original image
+   */
+  originalSrcset?: string
+
+  /**
+   * Url of thumbnail
+   */
+  thumbnail?: string
+
+  /**
+   * Width of original image
+   */
+  width?: string | number
+
+  /**
+   * Height of original image
+   */
+  height?: string | number
+
+  /**
+   * Alternate text for original image
+   */
+  alt?: string
+
+  /**
+   * Text for caption
+   */
+  caption?: string
+
+  /**
+   * Custom slide content
+   */
+  content?: JSX.Element
+
+  /**
+   * Custom slide content (raw html)
+   *
+   * TODO: deprecate, use `content` instead
+   */
+  html?: string
+
+  /**
+   * Item ID, for hash navigation
+   */
+  id?: string | number
+
+  /**
+   * Thumbnail is cropped
+   */
+  cropped?: boolean
+}
 export interface InternalAPI {
```

Comment:
@irudoy, now `InternalAPI` is able for import to lib users, but it seems, that there is no sense in it